internal/provider/provider.go

```diff
@@ -196,50 +240,147 @@ func (p *pingfederateProvider) Configure(ctx context.Context, req provider.Confi
 		}
 	}
 
-	// User must provide a username to the provider
+	// Check if the user has provided a username to the provider
 	var username string
-	if config.Username.IsUnknown() {
+	if config.Username.IsNull() {
+		username = os.Getenv("PINGFEDERATE_PROVIDER_USERNAME")
+	} else {
+		username = config.Username.ValueString()
+	}
+
+	// Check if the user has provided a password to the provider
+	var password string
+	if config.Password.IsNull() {
+		password = os.Getenv("PINGFEDERATE_PROVIDER_PASSWORD")
+	} else {
+		password = config.Password.ValueString()
+	}
+
+	// Check if the user has provided an access token to the provider
+	var accessToken string
+	if config.AccessToken.IsUnknown() {
```

Comment:
Why is the Unknown check removed for username & password?

Reply:
not sure, adding this back